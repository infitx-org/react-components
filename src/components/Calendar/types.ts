export type PossibleDate = Date | undefined;

export type PossibleDay = string | undefined;

export type Matrix = PossibleDay[][];

export type DisabledDays = (day: Date) => boolean;

export type DateRange = [PossibleDate, PossibleDate];
