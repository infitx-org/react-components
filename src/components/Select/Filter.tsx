import Search from "../../assets/icons/search-small.svg";
import Icon from "../Icon";
import { InputSize } from "../types";
import { getIconSizeByComponentSize } from "../shared";

interface FilterProps {
  size: `${InputSize}`;
}

function Filter({ open, size }: FilterProps) {
  return (
    <div className="mb-input__inner-icon input-select__icon">
      <Icon size={getIconSizeByComponentSize(size)} icon={<Search />} />
    </div>
  );
}

export default Filter;
