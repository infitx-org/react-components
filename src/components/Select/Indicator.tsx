import classnames from "classnames";
import Arrow from "../../assets/icons/arrow.svg";
import Icon from "../Icon";
import { InputSize } from "../types";

const indicatorSizes = {
  [InputSize.Small]: 9,
  [InputSize.Medium]: 10,
  [InputSize.Large]: 11,
};
interface IndicatorProps {
  open: boolean;
  size: `${InputSize}`;
}

function Indicator({ open, size }: IndicatorProps) {
  const className = classnames([
    "select__indicator",
    open && "select__indicator--open",
  ]);
  return (
    <Icon
      className={className}
      icon={<Arrow />}
      size={indicatorSizes[size]}
      fill="rgba(0,0,0,0.4)"
    />
  );
}

export default Indicator;
