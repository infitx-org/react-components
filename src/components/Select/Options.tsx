import React from "react";
import classnames from "classnames";
import "./Options.scss";
import Icon from "../Icon";
import { InputSize } from "../types";
import { getIconSizeByComponentSize } from "../shared";
import InfoSmall from "../../assets/icons/info-small.svg";
import CloseSmall from "../../assets/icons/close-small.svg";
// import ScrollBox from '../ScrollBox';
// import Tooltip from '../Tooltip';

/* eslint-disable jsx-a11y/no-noninteractive-tabindex */
/* eslint-disable jsx-a11y/tabindex-no-positive */

export type OptionValue = string | number | boolean;
type OptionLabel = string | number;

export type Option = {
  value: OptionValue;
  label: OptionLabel;
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  disabled?: boolean;
};

interface OptionProps {
  className?: string;
  size: `${InputSize}`;
  highlighted?: boolean;
  selected?: boolean;
  disabled?: boolean;
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  label: OptionLabel;
  onClick: (e: React.MouseEvent<HTMLDivElement>) => void;
}

function OptionItem({
  className,
  size,
  label,
  icon,
  selected = false,
  disabled = false,
  highlighted = false,
  onClick,
}: OptionProps) {
  const optionsClassNames = classnames([
    "input-select__options-item",
    size === InputSize.Small && "input-select__options-item--small",
    size === InputSize.Medium && "input-select__options-item--medium",
    size === InputSize.Large && "input-select__options-item--large",
    selected && "input-select__options-item--selected",
    disabled && "input-select__options-item--disabled",
    highlighted && "input-select__options-item--highlighted",
    className,
  ]);
  return (
    <div
      className={optionsClassNames}
      onClick={onClick}
      role="presentation"
      tabIndex={1}
    >
      {icon && (
        <Icon
          className="input-select__options-item__icon"
          icon={icon}
          size={getIconSizeByComponentSize(size)}
        />
      )}
      <div className="input-select__options-item__label">{label}</div>
    </div>
  );
}

interface OptionsProps {
  options: Option[];
  highlighted?: OptionValue;
  selected?: OptionValue;
  size: `${InputSize}`;
  clearable?: boolean;
  // maxHeight?: number;
  reverse?: boolean;
  onSelect: (option: Option) => void;
  onClear: () => void;
  optionsRef?: React.RefObject<HTMLDivElement>;
}

function Options({
  options,
  highlighted,
  selected,
  size,
  clearable,
  // maxHeight,
  reverse,
  onSelect,
  onClear,
  optionsRef,
}: OptionsProps) {
  const className = classnames([
    "input-select__options-wrapper",
    reverse && "input-select__options-wrapper--reverse",
    !reverse && "input-select__options-wrapper--regular",
    size === InputSize.Small &&
      reverse &&
      "input-select__options-wrapper--reverse-small",
    size === InputSize.Small &&
      !reverse &&
      "input-select__options-wrapper--regular-small",
    size === InputSize.Medium &&
      reverse &&
      "input-select__options-wrapper--reverse-medium",
    size === InputSize.Medium &&
      !reverse &&
      "input-select__options-wrapper--regular-medium",
    size === InputSize.Large &&
      reverse &&
      "input-select__options-wrapper--reverse-large",
    size === InputSize.Large &&
      !reverse &&
      "input-select__options-wrapper--regular-large",
  ]);

  let clearOption = null;
  if (clearable && selected !== undefined) {
    clearOption = (
      <OptionItem
        onClick={onClear}
        size={size}
        label="Clear"
        icon={<CloseSmall fill="#c33" />}
        className="input-select__options-item--clear"
      />
    );
  }

  let optionItems = null;
  if (options.length > 0) {
    optionItems = options.map((item, index) => (
      <OptionItem
        size={size}
        highlighted={item.value === highlighted}
        label={item.label}
        icon={item.icon}
        disabled={item.disabled === true}
        key={index.toString()}
        selected={selected === item.value}
        onClick={(e) => {
          e.preventDefault();
          e.stopPropagation();
          onSelect(item);
        }}
      />
    ));
  } else {
    optionItems = (
      <div className="input-select__options-item--no-options__box">
        <Icon
          icon={<InfoSmall fill="#999" />}
          size={getIconSizeByComponentSize(size)}
        />
        <div className="input-select__options-item--no-options__message">
          No options available
        </div>
      </div>
    );
  }
  return (
    <div className="input-select__options" ref={optionsRef}>
      <div
        className={className}
        style={{ maxHeight: "100px", overflow: "auto" }}
      >
        {/* <ScrollBox
          style={{ maxHeight }}
          handleStyle={{ borderRadius: "3px" }}
          trackStyle={{
            top: "2px",
            bottom: "2px",
            right: "4px",
            width: "5px",
          }}
          showTrack={false}
        > */}
        {clearOption}
        {optionItems}
        {/* </ScrollBox> */}
      </div>
    </div>
  );
}

export default Options;
