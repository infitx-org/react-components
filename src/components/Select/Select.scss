@use "../../scss/base/colors.scss" as *;
@use "../../scss/base/transitions.scss" as *;
@use "../../scss/components/fields.scss" as *;

.rc-select {
  @include rc-field-input;

  flex: 2 1 auto;
  vertical-align: middle;
  position: relative;
  height: 20px;
  background: transparent;
  border: none;
  outline: none;
  padding: 0;
  pointer-events: none;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: $color__dark-gray;
}

.rc-select--filtering {
  color: $color__blue--dark;
}
