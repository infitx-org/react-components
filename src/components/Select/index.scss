@use "./Select.scss";
@use "./components/Filter.scss";
@use "./components/Options.scss";
@use "../shared/Indicator.scss";
