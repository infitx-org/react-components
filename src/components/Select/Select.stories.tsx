import React from "react";
import Select from "./Select";

export default {
  title: "Library/Components/Select",
};

/* eslint-disable no-console */
const options = new Array(3).fill(0).map((_, index: number) => ({
  label: index + 1,
  value: index + 1,
}));

const Template = (args) => <Select {...args} options={options} />;

export const Default = Template.bind({});
Default.args = {
  size: "large",
  value: undefined,
  className: undefined,
  placeholder: "Choose a value",
  required: false,
  invalid: false,
  pending: false,
  options,
  onChange: console.log,
  onClear: console.log,
};

export const SelectedValue = Template.bind({});
SelectedValue.args = {
  value: options[0].value,
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};

export const Placeholder = Template.bind({});
Placeholder.args = {
  placeholder: "Select a value",
};

export const Pending = Template.bind({});
Pending.args = {
  pending: true,
};

export const Required = Template.bind({});
Required.args = {
  required: true,
};

export const Invalid = Template.bind({});
Invalid.args = {
  invalid: true,
};

export const Small = Template.bind({});
Small.args = {
  size: "small",
};

export const Medium = Template.bind({});
Medium.args = {
  size: "medium",
};

export const Large = Template.bind({});
Large.args = {
  size: "large",
};

export const OnClear = Template.bind({});
OnClear.args = {
  onClear: console.log,
};
