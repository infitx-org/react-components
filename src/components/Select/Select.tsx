import React from "react";
import find from "lodash/find";
import findIndex from "lodash/findIndex";
import classnames from "classnames";
import { InputSize } from "types";
import { KeyCodes } from "utils/keyCodes";
import mergeRefs from "utils/mergeRefs";
import Field, { Loader, Placeholder } from "../Field";
import Indicator from "./components/Indicator";
import Options, { Option, OptionValue } from "./components/Options";
import Filter from "./components/Filter";
import "./Select.scss";

export interface SelectProps
  extends Omit<React.SelectHTMLAttributes<HTMLInputElement>, "size" | "value"> {
  size?: `${InputSize}`;
  value?: OptionValue;
  className?: string;
  placeholder?: string;
  required?: boolean;
  pending?: boolean;
  options: Option[];
  onClear?: () => void;
}

export default React.forwardRef(function Select(
  {
    size = InputSize.Large,
    value,
    className,
    placeholder,
    required,
    pending,
    options = [],
    onClear,
    ...props
  }: SelectProps,
  ref: React.ForwardedRef<HTMLInputElement>
): JSX.Element {
  const inputRef = React.useRef<HTMLInputElement>(null);
  const optionsRef = React.useRef<HTMLDivElement>(null);
  const [selectedValue, setSelected] = React.useState<OptionValue | undefined>(
    value
  );
  const [highlighted, setHighlighted] = React.useState<
    OptionValue | undefined
  >();
  const [open, setOpen] = React.useState(false);
  const [focused, setFocused] = React.useState(false);
  const [filter, setFilter] = React.useState<string | undefined>(undefined);

  function enter() {
    setFocused(true);
    setOpen(true);
    setFilter(undefined);
    inputRef.current?.focus();
  }

  function leave() {
    setFocused(false);
    setOpen(false);
    setFilter(undefined);
  }

  function onSelect(newValue: OptionValue) {
    setSelected(newValue);
    setFilter(undefined);
    setOpen(false);
    inputRef.current?.focus();
  }

  function onFocus(e: React.FocusEvent<HTMLInputElement>) {
    if (!focused) {
      enter();
      props.onFocus?.(e);
    }
  }

  function onBlur(e: React.FocusEvent<HTMLInputElement>) {
    if (!focused) {
      props.onBlur?.(e);
    }
  }

  function onClearClick() {
    setSelected(undefined);
    onClear?.();
  }

  function onFieldClick(e: React.MouseEvent<HTMLDivElement>): void {
    e.preventDefault();
    e.stopPropagation();
    if (e.target === inputRef.current) {
      return;
    }
    if (open) {
      leave();
    } else {
      enter();
    }
  }

  function hasFilter(filterValue: string | undefined): boolean {
    return filterValue !== undefined && filterValue !== "";
  }

  function getOptions() {
    if (filter === undefined || filter === "") {
      return options;
    }
    const lowerCaseFilter = filter.toLowerCase();
    return options.filter((option: Option) =>
      option.label.toString().toLowerCase().includes(lowerCaseFilter)
    );
  }

  function scrollToOption(optionValue: OptionValue) {
    const filteredOptions = getOptions();
    const index = findIndex(filteredOptions, { value: optionValue });
    const optionDivs = document.querySelectorAll(".rc-select__options-item");
    const nextOption = optionDivs[index];

    if (nextOption) {
      (nextOption as HTMLDivElement).focus();
    }
    inputRef.current?.focus();
  }

  function highlightNextOption(next: boolean = true) {
    const filteredOptions = getOptions();
    const filteredOptionsLength = filteredOptions.length;
    const referenceValue = highlighted || selectedValue;
    let currentIndex = findIndex(filteredOptions, { value: referenceValue });

    function getNextEnabledOption(): OptionValue | undefined {
      let nextIndex = (currentIndex + (next ? 1 : -1)) % filteredOptionsLength;
      if (nextIndex < 0) {
        nextIndex = filteredOptions.length - 1;
      }
      currentIndex = nextIndex;
      const nextOption = filteredOptions[nextIndex];

      if (nextOption?.disabled) {
        return;
      }
      // eslint-disable-next-line
      return nextOption?.value;
    }

    let highlightedValue;
    let tries = filteredOptionsLength;
    while (!highlightedValue && tries >= 0) {
      highlightedValue = getNextEnabledOption();
      tries -= 1;
    }

    if (highlightedValue) {
      scrollToOption(highlightedValue);
    }
    setHighlighted(highlightedValue);
  }

  function handleArrows(e: React.KeyboardEvent<HTMLInputElement>) {
    const { keyCode } = e;

    if (keyCode === KeyCodes.Tab) {
      leave();
      return;
    }
    if (keyCode === KeyCodes.Up || keyCode === KeyCodes.Down) {
      e.preventDefault();
      highlightNextOption(keyCode === KeyCodes.Down);
      return;
    }
    if (keyCode === KeyCodes.Return) {
      e.preventDefault();
      if (open) {
        if (highlighted) {
          onSelect(highlighted);
        }
      } else {
        setOpen(true);
      }
    }
  }

  const filteredOptions = getOptions();
  const selectedItem = find(filteredOptions, { value: selectedValue });
  const selectedLabel = selectedItem?.label;

  const selectClassName = classnames([
    "rc-select",
    hasFilter(filter) && "rc-select--filtering",
  ]);

  return (
    <Field
      className={className}
      required={required && selectedValue === undefined}
      pending={pending}
      disabled={props.disabled}
      focused={focused}
      onClick={onFieldClick}
      onClickOutside={leave}
    >
      {placeholder && (
        <Placeholder
          label={placeholder}
          active={!!selectedLabel || focused}
          size={size}
        />
      )}
      <input
        {...props}
        className={selectClassName}
        type="text"
        ref={mergeRefs(ref, inputRef)}
        onFocus={onFocus}
        onBlur={onBlur}
        onChange={(e) => setFilter(e.target.value)}
        onKeyDown={handleArrows}
        value={filter !== undefined ? filter : selectedLabel || ""}
      />
      {filter !== undefined && <Filter size={size} />}
      {pending && <Loader size={size} />}
      <Indicator open={open} size={size} />
      {open && (
        <Options
          size={size}
          options={filteredOptions}
          selected={selectedValue}
          highlighted={highlighted}
          clearable={onClear !== undefined}
          onClear={onClearClick}
          onSelect={(option: Option) => onSelect(option.value)}
          optionsRef={optionsRef}
        />
      )}
    </Field>
  );
});
