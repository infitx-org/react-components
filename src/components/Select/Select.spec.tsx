import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import { InputSize } from "types";
import Select from "./Select";

const options = new Array(5).fill(0).map((_, index) => ({
  label: (index + 1).toString(),
  value: (index + 1).toString(),
}));

const commonProps = {
  options,
  onChange: jest.fn(),
};

function getOptions(container: HTMLElement): NodeListOf<HTMLDivElement> {
  return container.querySelectorAll(
    ".rc-select__option"
  ) as NodeListOf<HTMLDivElement>;
}
function getInput(container: HTMLElement): HTMLInputElement {
  return container.querySelector("input[type='text']") as HTMLInputElement;
}

describe("tests the select props", () => {
  it("renders the select", () => {
    const { container } = render(<Select {...commonProps} />);
    expect(container.querySelector(".rc-select")).toBeTruthy();
    expect(container.querySelectorAll('input[type="text"]')).toHaveLength(1);
  });

  it("does not renders options are closed", () => {
    const { container } = render(<Select {...commonProps} />);
    expect(container.querySelector(".rc-select__options")).toBeFalsy();
  });

  it("renders the label for the option value", () => {
    const [{ label, value }] = options;
    const { container } = render(<Select {...commonProps} value={value} />);
    const input = getInput(container);
    expect(input.value).toBe(label);
  });

  it("renders no label when option value does not exist", () => {
    const { container } = render(<Select {...commonProps} value="null" />);
    const input = getInput(container);
    expect(input.value).toBe("");
  });

  it("renders the placeholder", () => {
    const { container } = render(
      <Select {...commonProps} placeholder="test" />
    );
    expect(container.querySelector(".placeholder")).toBeTruthy();
  });

  it("renders the classname", () => {
    const { container } = render(<Select {...commonProps} className="test" />);
    expect(container.querySelector(".test")).toBeTruthy();
  });

  it("renders as disabled", () => {
    const { container } = render(<Select {...commonProps} disabled />);
    const input = getInput(container);
    expect(input.disabled).toBeTruthy();
  });

  it("renders as pending", () => {
    const { container } = render(<Select {...commonProps} pending />);
    expect(container.querySelector(".rc-spinner")).toBeTruthy();
  });

  it("renders as required", () => {
    const { container } = render(<Select {...commonProps} required />);
    expect(container.querySelector(".rc-field--required")).toBeTruthy();
  });

  it("renders as invalid", () => {
    const { container } = render(<Select {...commonProps} invalid />);
    expect(container.querySelector(".rc-field--invalid")).toBeTruthy();
    expect(container.querySelector(".rc-invalid-icon")).toBeTruthy();
  });

  it("renders the small, medium, large sizes", () => {
    Object.values(InputSize).forEach((size) => {
      const { container } = render(<Select {...commonProps} size={size} />);
      expect(container.querySelector(`.rc-field--${size}`)).toBeTruthy();
    });
  });

  it("renders the options when clicked", () => {
    const { container } = render(<Select {...commonProps} />);
    userEvent.click(getInput(container));
    expect(container.querySelector(".rc-select__options")).toBeInTheDocument();
  });

  it("renders the options when focused", () => {
    const { container } = render(<Select {...commonProps} />);
    fireEvent.focus(getInput(container));
    expect(container.querySelector(".rc-select__options")).toBeInTheDocument();
  });

  it("renders the clear option", () => {
    const { container } = render(
      <Select {...commonProps} onClear={jest.fn()} value={1} />
    );
    fireEvent.focus(getInput(container));
    expect(
      container.querySelector(".rc-select__option--clear")
    ).toBeInTheDocument();
  });

  it("selects a value when clicking the option", () => {
    const { container } = render(<Select {...commonProps} />);
    userEvent.click(getInput(container));
    const [option] = getOptions(container);
    userEvent.click(option);
    expect(getInput(container).value).toBe(options[0].label);
  });

  it("clear the value when clicking the clear option", () => {
    const { container } = render(
      <Select {...commonProps} onClear={jest.fn()} value={1} />
    );
    userEvent.click(getInput(container));
    userEvent.click(
      container.querySelector(".rc-select__option--clear") as Element
    );
    expect(getInput(container).value).toBe("");
  });

  it("triggers onClear when clicking the clear option", () => {
    const mockFn = jest.fn();
    const { container } = render(
      <Select {...commonProps} onClear={mockFn} value={1} />
    );
    userEvent.click(getInput(container));
    userEvent.click(
      container.querySelector(".rc-select__option--clear") as Element
    );
    expect(mockFn).toHaveBeenCalled();
  });

  it("triggers onFocus when focusing", () => {
    const mockFn = jest.fn();
    const { container } = render(<Select {...commonProps} onFocus={mockFn} />);
    fireEvent.focus(getInput(container));
    expect(mockFn).toHaveBeenCalled();
  });

  it("triggers onFocus when clicking", () => {
    const mockFn = jest.fn();
    const { container } = render(<Select {...commonProps} onFocus={mockFn} />);
    userEvent.click(getInput(container));
    expect(mockFn).toHaveBeenCalled();
  });

  it("triggers onFocus when tabbing", () => {
    const mockFn = jest.fn();
    render(<Select {...commonProps} onFocus={mockFn} />);
    userEvent.tab();
    expect(mockFn).toHaveBeenCalled();
  });

  it("triggers onBlur when clicking out", () => {
    const mockFn = jest.fn();
    const { container } = render(<Select {...commonProps} onBlur={mockFn} />);
    userEvent.click(getInput(container));
    userEvent.click(document.body);
    expect(mockFn).toHaveBeenCalled();
  });

  // it("triggers onBlur when tabbing away", () => {
  //   const mockFn = jest.fn();
  //   const { container } = render(<Select {...commonProps} onBlur={mockFn} />);
  //   userEvent.click(getInput(container));
  //   userEvent.tab();
  //   expect(mockFn).toHaveBeenCalled();
  // });

  it("triggers onChange when selecting a value", () => {
    const mockEvent = jest.fn();
    const { container } = render(
      <Select onChange={mockEvent} options={options} />
    );
    userEvent.click(getInput(container));
    const [option] = getOptions(container);
    userEvent.click(option);
    expect(mockEvent).toHaveBeenCalledWith("1");
  });

  // it("filter the options", () => {
  //   const [option] = options;
  //   const { container } = render(
  //     <Select options={options} value={option.value} />
  //   );
  //   userEvent.click(getInput(container));
  //   fireEvent.change(getInput(container), { target: { value: "1" } });
  //   const filteredOptions = container.querySelectorAll(".rc-select__option");
  //   expect(filteredOptions).toHaveLength(1);
  // });

  it("highlights the next option when pressing arrow down", () => {
    const [option] = options;
    const { container } = render(
      <Select options={options} value={option.value} />
    );
    userEvent.click(getInput(container));
    fireEvent.keyDown(getInput(container), { keyCode: 40 });
    const highlighted = container.querySelector(
      ".rc-select__option--highlighted"
    );
    expect(highlighted).toHaveTextContent("2");
  });

  it("highlights the next option when pressing arrow up", () => {
    const [option] = options;
    const { container } = render(
      <Select options={options} value={option.value} />
    );
    userEvent.click(getInput(container));
    fireEvent.keyDown(getInput(container), { keyCode: 38 });
    const highlighted = container.querySelector(
      ".rc-select__option--highlighted"
    );
    expect(highlighted).toHaveTextContent("5");
  });
});

// describe('Tests highlighiting with filtering', () => {
//   let wrapper;
//   const testOptions = [11, 22, 23, 24, 25].map(v => ({
//     label: v.toString(),
//     value: v,
//     disabled: v % 2 === 0,
//   }));

//   beforeEach(() => {
//     const mockEvent = jest.fn();
//     wrapper = mount(<Select onChange={mockEvent} options={testOptions} />);
//     wrapper.find('input[type="text"]').simulate('click');
//   });

//   it('highlights the correct option when pressing arrow down once', () => {
//     wrapper.find('input[type="text"]').simulate('keydown', { keyCode: 40 });
//     expect(wrapper.state('highlightedOption')).toEqual(11);
//   });

//   it('highlights the last available option when pressing arrow up once on filtered items', () => {
//     wrapper.find('input[type="text"]').simulate('keydown', { keyCode: 38 });
//     expect(wrapper.state('highlightedOption')).toEqual(25);
//   });

//   it('highlights the next available option when pressing arrow down once on filtered items', () => {
//     wrapper
//       .find('input[type="text"]')
//       .simulate('change', { target: { value: '2' } })
//       .simulate('keydown', { keyCode: 40 });
//     expect(wrapper.state('highlightedOption')).toEqual(23);
//   });

//   it('highlights the previous available option when pressing arrow up twice on filtered items', () => {
//     wrapper
//       .find('input[type="text"]')
//       .simulate('change', { target: { value: '2' } })
//       .simulate('keydown', { keyCode: 38 })
//       .simulate('keydown', { keyCode: 38 });
//     expect(wrapper.state('highlightedOption')).toEqual(23);
//   });
// });

// it('renders the Select correctly when multiple props are set', () => {
//   const wrapper = shallow(<Select value="value-1" id="test-id" options={options} />);
//   expect(shallowToJson(wrapper)).toMatchSnapshot();
// });

// it('renders the validation wrapper', () => {
//   const wrapper = shallow(<Select placeholder="test-Select" />);
//   expect(wrapper.find(ValidationWrapper)).toHaveLength(1);
// });

// it('sorts the options by value ascending', () => {
//   const reverseOptions = [...options].reverse();
//   const wrapper = mount(<Select selected="value-1" options={reverseOptions} sortBy="value" />);
//   expect(wrapper.state().options[0]).toBe(options[0]);
// });

// it('sorts the options by value descending', () => {
//   const wrapper = mount(
//     <Select selected="value-1" options={options} sortBy="value" sortAsc={false} />,
//   );
//   const lastOption = options[options.length - 1];
//   expect(wrapper.state().options[0]).toBe(lastOption);
// });

// it('sorts the options by label', () => {
//   const unsortedOptions = toSelectOptions([1, 9, 2, 8, 5]);
//   const wrapper = mount(<Select selected="value-1" options={unsortedOptions} sortBy="label" />);
//   expect(wrapper.state().options[0].label).toBe('1');
// });

// it('sorts the options by disabled', () => {
//   const unsortedOptions = toSelectOptions([1, 9, 2, 8, 5]);
//   unsortedOptions[2].disabled = true;
//   unsortedOptions[4].disabled = true;
//   const wrapper = mount(<Select selected="value-1" options={unsortedOptions} sortBy="disabled" />);
//   expect(wrapper.state().options[0].disabled).toBe(true);
//   expect(wrapper.state().options[1].disabled).toBe(true);
// });
