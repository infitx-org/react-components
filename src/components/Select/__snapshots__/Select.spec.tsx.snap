// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Select correctly when multiple props are set 1`] = `
<div>
  <div
    class="rc-field rc-field--primary rc-field--large rc-field--large"
    role="presentation"
  >
    <label
      class="placeholder placeholder--large"
    >
      test
    </label>
    <input
      class="rc-select rc-select--large"
      type="text"
      value=""
    />
    <svg
      class="rc-select__indicator"
      fill="rgba(0,0,0,0.4)"
      height="11"
      stroke="transparent"
      width="11"
    />
  </div>
</div>
`;
